### ISSUE TITLE
Access docker hub with error “x509: certificate signed by unknown authority”, when the self-signed certificate is emloyed

### ISSUE THREAD

**Issue by wuxiaojie** [docker/docker#15227]
When I construct docker private registry using nginx with self-signed certificate, to access the  private registry,  the self-signed root certificate is need be added in the client as following:
``` bash 
    $ cat cacert.pem | sudo tee -a /etc/ssl/certs/ca-certificates.crt
```
Then, I successfully access the private registry. However, when the client tries to access docker hub, there exists a error:
``` bash
    Get https://index.docker.io/v1/repositories/library/centos/images: x509: 
certificate signed by unknown authority
```
There exist some solutions to solve this issue, but it needs to remove ca-certificates.crt from the client and restart docker daemon.  I want to access docker hub more automatically without restart, but I have no any idea. Look forwarding to your helps, thanks!

     /etc/ssl/certs/ca-certificates.crt is created by my self-created root certificate

**Comment by wuxiaojie** [comment]
fixes #15227
fixed #15227
fix #15227
closes#15227
close #15227
closed #15227

**Comment by wuxiaojie** [comment]
fixes #15227
fixed #15227
fix #15227
closes #15227
close #15227
closed #15227
